src/plugins/kibana/public/settings/sections/advanced/lib/is_immutable_config.js

```diff
@@ -0,0 +1,15 @@
+define(function (require) {
+  var _ = require('lodash');
+
+  var IMMUTABLE_CONFIG_VALS = ['buildNum'];
```

Comment:
What do you think about adding buildNum to the defaults file (which is more of a schema file at this point) and giving it a readOnly attribute, or something to that effect? It would probably be easier to manage the the schema if all of the field properties were listed in one place, rather than in defaults and is_mutable_config.js.